packages/next/src/build/webpack/loaders/next-flight-loader/action-proxy.ts

```diff
@@ -1,10 +1,24 @@
-export default function ensureServerEntryExports(actions: any[]) {
-  for (let i = 0; i < actions.length; i++) {
-    const action = actions[i]
-    if (typeof action !== 'function') {
-      throw new Error(
-        `A "use server" file can only export async functions, found ${typeof action}.`
-      )
+export default function createActionProxy(action: any, originalAction: any) {
+  action.bind = function (_: any, ...boundArgs: any[]) {
+    const newAction = async function (...args: any[]) {
+      if (originalAction) {
+        return originalAction(newAction.$$bound.concat(args))
+      } else {
+        // In this case we're calling the user-defined action directly.
+        return action(...newAction.$$bound, ...args)
+      }
     }
+
+    for (const key of Object.keys(action)) {
```

Comment:
You know what keys are valid to transfer here so might as well explicitly list them out which is better for perf too.

Reply:
I saw things like `$$FORM_ACTION` being added by `createServerReference`, so I'm worried about unknown effects. But sure I'll get a list for it.

Reply:
`$$FORM_ACTION` is only added in the Client Graph where `createServerReference` is used - and this is in the Server Graph. Since it's only SSR that renders those as forms.

Reply:
The case I was thinking is that people could still have actions imported in the Client Graph and then `bind` it to create a new action.

Reply:
For that case we should just add it in the existing `createServerReference` which already covers all of those.

Reply:
We should probably just add the whole `__create_action_proxy__` helper upstream in React itself so that it works the same way.